connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -116,6 +118,20 @@
         }
     }
 
+
+    /**
+     * Extract headers from REST call and add to client request
+     * @param headers
+     * @param req
+     */
+    private static void addHeadersToRequest(HttpHeaders headers, Request req) {
+        String credentialAuthorization = headers != null ? headers.getHeaderString("Authorization") : null;
+        if (credentialAuthorization != null) {
+            req.header("Authorization", credentialAuthorization);
+        }
+    }
```

Comment:
It's probably better to have nested if-blocks to simplify things and make it more clear when the header is added to the request:
```
if (headers != null) {
    String authorization = headers.getHeaderString("Authorization");
    if (authorization != null) {
        req.header("Authorization", authorization);
    }
}
```

Reply:
Also, we have "Authorization" here twice -- probably worthwhile adding a constant just to make sure we don't have a bug in the future due to a typo.

Reply:
Fixed!